network-dashboard-proxy/src/main.rs

```diff
@@ -0,0 +1,253 @@
+            serde_json::json!({
+                "mintDistribution": cp.mint_distribution,
+                "transactionFeeDistribution": cp.transaction_fee_distribution,
+                "gasRewards": cp.gas_rewards
+            })
+        }
+        v2::ChainParameters::V1(cp) => {
+            serde_json::json!({
+                "mintDistribution": cp.mint_distribution,
+                "transactionFeeDistribution": cp.transaction_fee_distribution,
+                "gasRewards": cp.gas_rewards
+            })
+        }
+    };
+    Ok(serde_json::json!({
+        "finalizationData": finalization_data,
+        "transactionSummaries": txs,
+        "specialEvents": special,
+        "pendingUpdates": pending_updates,
+        "chainParameters": {
+            "rewardParameters": reward_parameters
+        }
+    })
+    .into())
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let app = App::parse();
+
+    {
+        use tracing_subscriber::prelude::*;
+        let log_filter = tracing_subscriber::filter::Targets::new()
+            .with_target(module_path!(), app.log_level)
+            .with_target("tower_http", app.log_level);
+        tracing_subscriber::registry()
+            .with(tracing_subscriber::fmt::layer())
+            .with(log_filter)
+            .init();
+    }
+
+    let endpoint = if app
+        .endpoint
+        .uri()
+        .scheme()
+        .map_or(false, |x| x == &http::uri::Scheme::HTTPS)
+    {
+        app.endpoint
+            .tls_config(ClientTlsConfig::new())
+            .context("Unable to construct TLS configuration for Concordium API.")?
+    } else {
+        app.endpoint
+    }
+    .connect_timeout(std::time::Duration::from_secs(10))
+    .timeout(std::time::Duration::from_millis(app.request_timeout));
+
+    let client = v2::Client::new(endpoint)
+        .await
+        .context("Unable to establish connection to the node.")?;
+
+    // build routes
+    let server = Router::new().route("/v1/transactionStatus", get(transaction_status))
+        .route("/v1/consensusStatus", get(consensus_status))
+        .route("/v1/blockSummary/:blockHash", get(block_summary))
+        .route("/v1/blockInfo/:blockHash", get(block_info))
+        .route("/v1/blocksByHeight/:height", get(blocks_by_height))
+        .with_state(client)
+        .layer(tower_http::trace::TraceLayer::new_for_http().
+               make_span_with(DefaultMakeSpan::new().
+                              include_headers(app.log_headers)).
+               on_response(DefaultOnResponse::new().
+                           include_headers(app.log_headers)))
+        .layer(tower_http::timeout::TimeoutLayer::new(
+            std::time::Duration::from_millis(app.request_timeout),
+        ))
+        .layer(tower_http::limit::RequestBodyLimitLayer::new(0)) // no bodies, we only have GET requests.
+        .layer(tower_http::cors::CorsLayer::permissive().allow_methods([http::Method::GET]));
```

Comment:
I understand why this is needed for development, but is it needed in production?
We don't have sensitive information, so it is not that big of a deal, I guess

Reply:
It might not be needed since it is all routed behind a single URL as I see it.

But I also don't see anything wrong with this, since we only allow GET, and in any case there is absolutely nothing sensitive here.

So to avoid any annoying issues I'll just keep it.